R/dsl.R

```diff
@@ -10,59 +10,21 @@ uiicon <- function(type = "", ...) {
   shiny::tags$i(class = paste(type, "icon"), ...)
 }
 
-#' Create dropdown Semantic UI component
-#'
-#' This creates a default dropdown using Semantic UI styles with Shiny input. Dropdown is already initialized
-#' and available under input[[name]].
-#'
-#' @param name Input name. Reactive value is available under input[[name]].
-#' @param choices All available options one can select from.
-#' @param choices_value What reactive value should be used for corresponding choice.
-#' @param default_text Text to be visible on dropdown when nothing is selected.
-#' @param value Pass value if you want to initialize selection for dropdown.
-#'
-#' @examples
-#' ## Only run examples in interactive R sessions
-#' if (interactive()) {
-#'
-#'   library(shiny)
-#'   library(shiny.semantic)
-#'   ui <- function() {
-#'       shinyUI(
-#'         semanticPage(
-#'           title = "Dropdown example",
-#'           suppressDependencies("bootstrap"),
-#'           uiOutput("dropdown"),
-#'           p("Selected letter:"),
-#'           textOutput("selected_letter")
-#'        )
-#'      )
-#'   }
-#'   server <- shinyServer(function(input, output) {
-#'      output$dropdown <- renderUI({
-#'          dropdown("simple_dropdown", LETTERS, value = "A")
-#'      })
-#'      output$selected_letter <- renderText(input[["simple_dropdown"]])
-#'   })
-#'
-#'   shinyApp(ui = ui(), server = server)
-#' }
-#'
-#' @export
-dropdown <- function(name, choices, choices_value = choices, default_text = 'Select', value = NULL) {
-  unique_dropdown_class <- paste0('dropdown_name_', name)
+icon_if_present <- function(icon = "") {
+  if(icon != "") {
+    uiicon(icon)
+  }
+}
 
-  shiny::tagList(
-    shiny::div(class = paste("ui selection fluid dropdown", unique_dropdown_class),
-               shiny_text_input(name, shiny::tags$input(type = "hidden", name = name), value = value),
-               uiicon("dropdown"),
-               shiny::div(class = "default text", default_text),
-               shiny::div(class = "menu",
-                          purrr::map2(choices, choices_value, ~ div(class = "item", `data-value` = .y, .x))
-               )
-    ),
-    shiny::tags$script(paste0("$('.ui.dropdown.", unique_dropdown_class, "').dropdown().dropdown('set selected', '", value,"');"))
-  )
+#' Create Semantic UI label tag
+#' 
+#' This creates a label tag using Semantic UI.
+#' 
+#' @param type Type of the label. Look at https://semantic-ui.com/elements/label.html for all possibilities.
+#' @param ... Other arguments to be added such as content of the tag (text, icons) and/or attributes (style)
+uilabel <- function(type = "", ...) {
+  div(class = paste("ui label", type),
```

Comment:
Many Semantic UI examples contain linkable labels (replacing `div` with `a`), maybe we could add this feature too in here?